lib/ClusterShell/Gateway.py

```diff
@@ -343,6 +343,10 @@ def gateway_main():
     task.set_default("stdout_msgtree", False)
     task.set_default("stderr_msgtree", False)
 
+    if sys.stdin is None:
```

Comment:
`sys.stdin` is used earlier above, so that actually won't work. Either put the check above, or leave the Gateway as is for now (I'm fine either way).